include/LightGBM/dataset.h

```diff
@@ -402,6 +408,12 @@ class Dataset {
         const int group = feature2group_[feature_idx];
         const int sub_feature = feature2subfeature_[feature_idx];
         feature_groups_[group]->PushData(tid, sub_feature, row_idx, inner_data.second);
+        if (has_raw_) {
```

Comment:
Shall we consider store the raw feature value in the PushOneData method of Dataset, which is called in c_api.cpp for CSC format data?

Reply:
I think I've implemented what you're suggesting, but let me know if I misunderstood.